test/unit/rules/lint-require-button-type-test.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+const generateRuleTests = require('../../helpers/rule-test-harness');
+
+generateRuleTests({
+  name: 'require-button-type',
+
+  config: true,
+
+  good: [
```

Comment:
Should add valid test cases to ensure this doesn't affect other elements:

```
<div></div>
<div type="randomType"></div>
```

Reply:
good idea, thanks :)